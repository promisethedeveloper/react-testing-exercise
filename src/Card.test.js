import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";
import image1 from "./image1.jpg";

// Smoke test
it("renders without crashing", () => {
	render(<Card />);
});

// Snapshot test
it("should match snapshots", () => {
	const { asFragment } = render(
		<Card
			caption="Photo by Richard Pasquarella on Unsplash"
			src={image1}
			currNum={1}
			totalNum={3}
		/>
	);
	expect(asFragment()).toMatchSnapshot();
});
