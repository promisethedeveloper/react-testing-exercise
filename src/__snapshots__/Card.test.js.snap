// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Richard Pasquarella on Unsplash
    </h4>
    <img
      alt="Photo by Richard Pasquarella on Unsplash"
      class="Card-image"
      src="image1.jpg"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;
